Modules/_decimal/_decimal.c

```diff
@@ -2514,7 +2514,7 @@ dec_from_float(PyObject *type, PyObject *pyfloat)
     CURRENT_CONTEXT(context);
     result = PyDecType_FromFloatExact(&PyDec_Type, pyfloat, context);
     if (type != (PyObject *)&PyDec_Type && result != NULL) {
-        Py_SETREF(result, PyObject_CallFunctionObjArgs(type, result, NULL));
+        Py_SETREF(result, _PyObject_CallOneArg(type, result));
```

Comment:
This is a code path that is practically never taken. Additionally, `PyDecType_FromFloatExact()` will completely dominate the run time, so any marginal speedup will not be noticed.

Please leave such changes to Module authors/maintainers.